bpo-41045: Add documentation for f-string's self documenting expressions

rishi93 opened the pull request:
Missing documentation describing the = specifier for f-strings has been added

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-41045](https://bugs.python.org/issue41045) -->
https://bugs.python.org/issue41045
<!-- /issue-number -->

rishi93 committed:
bpo-41045: Add documentation for f-string's self documenting expressions

Missing documentation describing the = specifier for f-strings has been added

ericvsmith commented on Doc/reference/lexical_analysis.rst:
```diff
@@ -758,6 +758,16 @@ include expressions.
    >>> foo.__doc__ is None
    True
 
+F-strings can also help with "printf-style" debugging.
```
I think this needs to at least describe the feature. See my 2 points in bpo-41045 for what I think needs to be included.

rishi93 committed:
bpo-41045: Add documentation for f-string's self documenting expressions

Added more examples to explain f-string's self documenting expressions when format specifiers are used

rishi93 committed:
bpo-41045: Add documentation for f-string's self documenting expressions

Corrected typos with single and double quotes in the f-string's examples

ericvsmith reviewed (commented):
I think this still needs some polishing, but I'm not very good at writing documentation. Hopefully someone else will respond with some suggestions.

amaajemyfren reviewed (commented):
I think the line
```reSt
replacement_field: "{" `f_expression` ["!" `conversion`] [":" `format_spec`] "}"
```
Should be changed to 
```reSt
replacement_field: "{" `f_expression` ["="] ["!" `conversion`] [":" `format_spec`] "}"
```

amaajemyfren commented on Doc/reference/lexical_analysis.rst:
```diff
@@ -758,6 +758,30 @@ include expressions.
    >>> foo.__doc__ is None
    True
 
+F-strings can also help with "printf-style" debugging.
```
Hey @rishi93,

Would you consider adding that this is new from version 3.8?

amaajemyfren commented on Doc/reference/lexical_analysis.rst:
```diff
@@ -758,6 +758,30 @@ include expressions.
+Optional format specifiers can be placed after the equals sign.
+If no format specifiers are used then :meth:`repr` is used on the expression.
+If format specifier (like :20) is given, then :meth:`str` is used on the
+expression. You can use the repr() with the !r format specifier
```
Hey @rishi93,

I think ```!r``` is a conversion not a format specifier.
